website/project/model.py

```diff
@@ -3479,6 +3419,204 @@ def require_approval(self, user, notify_initiator_on_complete=False):
+        if not user.is_affiliated_with_institution(inst):
+            raise UserNotAffiliatedError('User is not affiliated with {}'.format(inst.name))
+        if inst.node == self.primary_institution:
+            return False
+        previous = Institution(self.primary_institution) if self.primary_institution else None
+        self.primary_institution = inst
+        if inst.node not in self.affiliated_institutions:
+            self.affiliated_institutions.append(inst)
+        if log:
+            self.add_log(
+                action=NodeLog.PRIMARY_INSTITUTION_CHANGED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    },
+                    'previous_institution': {
+                        'id': previous._id if previous else None,
+                        'name': previous.name if previous else 'None'
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def remove_primary_institution(self, user, log=True):
+        inst = self.primary_institution
+        if not inst:
+            return False
+        self.primary_institution = None
+        if inst in self.affiliated_institutions:
+            self.affiliated_institutions.remove(inst)
+        inst = Institution(inst)
+        if log:
+            self.add_log(
+                action=NodeLog.PRIMARY_INSTITUTION_REMOVED,
+                params={
+                    'node': self._primary_key,
+                    'institution': {
+                        'id': inst._id,
+                        'name': inst.name
+                    }
+                },
+                auth=Auth(user)
+            )
+        return True
+
+    def institution_url(self):
+        return self.absolute_api_v2_url + 'institution/'
+
+    def institution_relationship_url(self):
+        return self.absolute_api_v2_url + 'relationships/institution/'
+
+class SpecialList(list):
+    def __init__(self, node, private_target, init=None):
+        super(SpecialList, self).__init__(init or [])
+        self.node = node
+        self.target = private_target
+
+    def append(self, to_append):
+        junk = getattr(self.node, self.target)
+        junk.append(to_append.node)
+        setattr(self.node, self.target, junk)
+
+class InstitutionQuerySet(MongoQuerySet):
+    def __init__(self, queryset):
+        super(InstitutionQuerySet, self).__init__(queryset.schema, queryset.data)
+
+    def __iter__(self):
+        for each in super(InstitutionQuerySet, self).__iter__():
+            yield Institution(each)
+
+    def _do_getitem(self, index):
+        item = super(InstitutionQuerySet, self)._do_getitem(index)
+        if isinstance(item, MongoQuerySet):
+            return self.__class__(item)
+        return Institution(item)
+
+class Institution():
```

Comment:
Documentation for this class and how it relates to the ```Node``` class would be awesome :tada: